internal/annotation/filter.go

```diff
@@ -138,60 +138,68 @@ func validateAnnotation(key string, truthy bool) error {
 // filter implements a filter for objects with a truthy "paused" annotation (see Key).
 // When this annotation is removed or value does not evaluate to "true",
 // the controller will see events from these objects again.
-type filter struct {
+type filter[T client.Object] struct {
 	key  string
 	ret  bool
 	log  logr.Logger
-	hdlr *handler.EnqueueRequestForObject
+	hdlr *handler.TypedEnqueueRequestForObject[T]
 }
 
 // Create implements predicate.Predicate.Create().
-func (f *filter) Create(evt event.CreateEvent) bool {
-	if evt.Object == nil {
+func (f *filter[T]) Create(evt event.TypedCreateEvent[T]) bool {
+	var obj client.Object = evt.Object
+	if obj == nil {
```

Comment:
This was needed to maintain the existing expectations of the API where a `nil` object in the event would result in returning a default filter value